File: app/src/main/java/com/maye/today/register/RegisterModelImpl.java
package com.maye.today.register;

import com.maye.today.domain.User;
import com.maye.today.network.RetrofitUtil;


import java.util.HashMap;
import java.util.Map;

import okhttp3.ResponseBody;
import rx.Observable;

/**
 * RegisterModel实现
 */
public class RegisterModelImpl implements RegisterModel {

    @Override
    public Observable<ResponseBody> register(User user) {

        Map<String, String> map = new HashMap<>();
        map.put("username", user.getUsername());
        map.put("nickname", user.getNickname());
        map.put("avater", user.getAvatar());
        map.put("password", user.getPassword());
        map.put("phone", user.getPhone());
        map.put("email", user.getEmail());

        return RetrofitUtil.registerServer().register(map);
    }

}

File: LoginModule/src/main/java/com/maye/register/mvp/RegisterModelImpl.kt
package com.maye.register.mvp

import com.maye.base.bean.User
import com.maye.net.NetUtil
import com.maye.register.api.RegisterServer
import io.reactivex.rxjava3.core.Observable
import okhttp3.ResponseBody
import java.util.*

/**
 * RegisterModel实现
 */
class RegisterModelImpl : RegisterModel {
    override fun register(user: User): Observable<ResponseBody> {
        val map: MutableMap<String, String> = HashMap()
        map["username"] = user.getUsername()
        map["nickname"] = user.getNickname()
        map["avater"] = user.getAvatar()
        map["password"] = user.getPassword()
        map["phone"] = user.getPhone()
        map["email"] = user.getEmail()
        return NetUtil.createApi(RegisterServer::class.java).register(map)
    }
}